agent/utils/mobypkgwrapper/Plugins.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package mobypkgwrapper
+
+import (
+	"fmt"
+	mobyplugins "github.com/docker/docker/pkg/plugins"
+)
+
+// Plugins wraps moby/pkg/plugins methods for testing
+type Plugins interface {
```

Comment:
Does it make sense to make the Scan a method of the `DockerClient`, so you don't need to create a new interface and also you don't need a new obj in agent struct?

Reply:
i was wondering the same thing, but i also think we're trying to keep this isolated so we can deprecate with minimal fuss.

Reply:
No DockerClient is a wrapper for go-dockerclient (fsouza/go-dockerclient repo), this is for Moby (moby/moby repo). I didn't want to mix them up, and we should be more careful of adding more moby/moby dependency.